### ISSUE TITLE
License

### ISSUE THREAD

**Issue by wallop-123** [mimani/mongoose-diff-history#45]
I am working on a project and would like to incorporate mongoose-diff-history in that project.
When looking at the license (GPL-2.0) it would require me to release the full source of my project.
I have no intention to do this with regards to my own private written code for this specific project.

Are there perhaps plans to switch to a different license which would allow me to use this library in this closed-source project?

**Comment by harikrishnakanchi** [comment]
I think GPL 2.0 requires you distribute your source only if you distribute your software. Find more info [here](https://choosealicense.com/licenses/gpl-2.0/)

**Comment by wallop-123** [comment]
Well it would be incorporated in commercial software which is licensed to customers and might be installed off-site. Therefore I'm not comfortable in using GPL 2.0 software in this case.

**Comment by zjr** [comment]
@mimani do you have any thoughts or feelings about this?

**Comment by wallop-123** [comment]
@mimani Would you please give your input to this?
I need to make a decision soon on which library to use and help develop further.
After reviewing this project it would really suit my needs with some small changes which would (in my opinion) benifit other users.

**Comment by mimani** [comment]
Hey @wallop-123 , 

I give you permission to use the library without the liability to release your source code. To do the due diligence we can change the licence for this library to more relaxed "Apache License 2.0" which does not have clause to share the source code.

However, it will be really helpful if you can provide your use-case for which you are using this library, if it possible/allowed at your end, which can be put in the users section in ReadMe for this library. Having examples of real-life usage will give more confidence to future users of this library. Thanks.

**Comment by jplusje** [comment]
Hello @mimani ,

I am @wallop-123, I created that account to seperate work/private and realized I asked the question under the wrong account, switching to "Apache License 2.0" would be perfect, it would allow me to continue working on this library to further support my use-case.

I already started work (just to investigate the possibilities of this library) some time ago and some of my preliminary work can be seen in this branch: https://github.com/jplusje/mongoose-diff-history/tree/multiple-connections. When everything works as expected I will create a PR to merge the library changes back to this repo.

I am going to use mongoose-diff-history to create a full audit trail for all objects within the application which is connected to multiple mongoose databases. I would be glad to add my use-case to the readme when I've progressed further.
